### DISCUSSION TITLE
Idea: Hypertext link label

### DISCUSSION THREAD

**Discussion by lupon1** [israel-dryer/ttkbootstrap#139] [Ideas]
A idea for a new widget type is a label with a link to open a website in the default browser.
Something like this:

![linkLabel](https://user-images.githubusercontent.com/49123184/149499039-cd52b3db-4dc7-4184-ab3b-52b7cfeb3a9e.gif)

```
import tkinter as tk
import ttkbootstrap as ttk
from webbrowser import open_new_tab

class linkLabel:
    def __init__(self, link, txt='', bootstyle='primary', **kwargs):
        if txt == '': txt = link
        fontFamily = tk.font.nametofont("TkDefaultFont").cget("family")
        fontSize = tk.font.nametofont("TkDefaultFont").cget("size")
        fontWeight = tk.font.nametofont("TkDefaultFont").cget("weight")
        self.font = tk.font.Font(family=fontFamily, size=fontSize, weight=fontWeight)
        self.label=ttk.Label(kwargs, bootstyle=bootstyle, text=txt, cursor='hand2')
        self.link = link
        for elem in ('<Enter>', '<Leave>', '<Button-1>'): self.label.bind(elem, self.action)
        
    def action(self, event):
        if event.type == '4':
            open_new_tab(self.link)
        else:
            if event.type == '7':
                self.font.configure(underline=True)
            elif event.type == '8':
                self.font.configure(underline=False)
            event.widget.configure(font=self.font)
        
    def pack(self, **kwargs):
        self.label.pack(kwargs)

win = tk.Tk()
win.title('ttkbootstrap Link Label Example')
lbl = linkLabel('http://www.google.com/', 'Google')
lbl.pack(padx=150, pady=100)
win.mainloop()
```

**Comment by israel-dryer** [comment]
I do have a link button style, but it doesn't apply font underlined or have this functionality. Thanks for the idea. I'll definitely consider adding something like this.

**Reply by NewbieXvwu** [reply]
I think its function should not only open links. It should be able to execute arbitrary code.

**Comment by TheDemonGuard** [comment]
@lupon1 
I have created a new feature request #796.
You can add any more information if necessary to this feature request.

You can also create new feature in the future, which might be a better way to get more attention and possibility of getting implemented.

> A idea for a new widget type is a label with a link to open a website in the default browser. Something like this:
> 
> ![linkLabel](https://user-images.githubusercontent.com/49123184/149499039-cd52b3db-4dc7-4184-ab3b-52b7cfeb3a9e.gif) [ ](https://user-images.githubusercontent.com/49123184/149499039-cd52b3db-4dc7-4184-ab3b-52b7cfeb3a9e.gif) ![linkLabel](https://user-images.githubusercontent.com/49123184/149499039-cd52b3db-4dc7-4184-ab3b-52b7cfeb3a9e.gif) [ ](https://user-images.githubusercontent.com/49123184/149499039-cd52b3db-4dc7-4184-ab3b-52b7cfeb3a9e.gif)
> 
> ```
> import tkinter as tk
> import ttkbootstrap as ttk
> from webbrowser import open_new_tab
> 
> class linkLabel:
>     def __init__(self, link, txt='', bootstyle='primary', **kwargs):
>         if txt == '': txt = link
>         fontFamily = tk.font.nametofont("TkDefaultFont").cget("family")
>         fontSize = tk.font.nametofont("TkDefaultFont").cget("size")
>         fontWeight = tk.font.nametofont("TkDefaultFont").cget("weight")
>         self.font = tk.font.Font(family=fontFamily, size=fontSize, weight=fontWeight)
>         self.label=ttk.Label(kwargs, bootstyle=bootstyle, text=txt, cursor='hand2')
>         self.link = link
>         for elem in ('<Enter>', '<Leave>', '<Button-1>'): self.label.bind(elem, self.action)
>         
>     def action(self, event):
>         if event.type == '4':
>             open_new_tab(self.link)
>         else:
>             if event.type == '7':
>                 self.font.configure(underline=True)
>             elif event.type == '8':
>                 self.font.configure(underline=False)
>             event.widget.configure(font=self.font)
>         
>     def pack(self, **kwargs):
>         self.label.pack(kwargs)
> 
> win = tk.Tk()
> win.title('ttkbootstrap Link Label Example')
> lbl = linkLabel('http://www.google.com/', 'Google')
> lbl.pack(padx=150, pady=100)
> win.mainloop()
> ```
